YouTubePlayer/YouTubePlayer.js

```diff
@@ -30,36 +42,66 @@ export default class {
     async clickHandler() {
         // Click handler is only needed once; remove it as soon as it's been used
         this.removeClickHandler();
-        // Empty preview to give the user a quick feedback
-        this.element.innerHTML = '';
+        this.addLoadingClass();
         await this.loadYouTubeAPI();
         this.displayAndPlayVideo();
     }
 
     async loadYouTubeAPI() {
         // YouTube script was already loaded, Player is ready
         if (window.YT && window.YT.Player && typeof window.YT.Player === 'function') return;
-        // Load YouTube iFrame API script
-        var tag = document.createElement('script');
-        tag.src = 'https://www.youtube.com/iframe_api';
-        var firstScriptTag = document.getElementsByTagName('script')[0];
-        firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+        // Check if there is already a YouTube script: If there is, just wait until it's done
+        const existingTag = document.querySelector(`script[src="${this.youTubeScriptURL}"]`)
+        // There is no script tag: Create and add it to the DOM
+        if (!existingTag) {
+            var tag = document.createElement('script');
+            tag.setAttribute('src', this.youTubeScriptURL);
+            // There must be a script somewhere as this code itself is a script
+            var firstScriptTag = document.getElementsByTagName('script')[0];
+            firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+        }
         return new Promise((resolve) => {
             window.onYouTubeIframeAPIReady = resolve;
         });
     }
 
+    addLoadingClass() {
+        if (this.getLoadingClass()) {
+            requestAnimationFrame(() => this.element.classList.add(this.getLoadingClass()));
+        }
+    }
+
+    removeLoadingClass() {
+        if (this.getLoadingClass()) {
+            requestAnimationFrame(() => this.element.classList.remove(this.getLoadingClass()));
+        }
+    }
+
+    /**
+     * Get name of class that should be added to element from element's attributes
+     * @return {string}
+     */
+    getLoadingClass() {
+        const className = this.element.getAttribute(this.loadingClassAttribute) || '';
+        if (!className) {
+            console.log(`YouTubePlayer: Attribute ${this.loadingClassAttribute} not set on element, cannot add loading class.`);
+        }
+        return className;
+    }
+
+
     displayAndPlayVideo() {
-        const videoId = this.element.getAttribute('video-id');
+        const videoId = this.element.getAttribute(this.videoIdAttribute);
```

Comment:
just a detail: better use https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes here? this.videoDatasetID = 'videoId'

Reply:
A very good point! Just learnt something, thanks! 🙇‍♂️